File: iOS/Pool/Sources/Pool/ManageGamblers/PoolMemberModel.swift
import Foundation
import DataPool

public struct PoolMemberModel: Identifiable, Hashable, Codable, Sendable {
    public let gamblerId: String
    public let gamblerUsername: String
    public let gamblerEmail: String
    public let isOwner: Bool

    public var id: String { gamblerId }

    public init(
        gamblerId: String,
        gamblerUsername: String,
        gamblerEmail: String,
        isOwner: Bool = false
    ) {
        self.gamblerId = gamblerId
        self.gamblerUsername = gamblerUsername
        self.gamblerEmail = gamblerEmail
        self.isOwner = isOwner
    }
}

extension PoolMember {
    func toPoolMemberModel() -> PoolMemberModel {
        PoolMemberModel(
            gamblerId: gamblerId,
            gamblerUsername: gamblerUsername,
            gamblerEmail: gamblerEmail,
            isOwner: isOwner
        )
    }
}

func poolMemberPlaceholderModel() -> PoolMemberModel {
    PoolMemberModel(
        gamblerId: UUID().uuidString,
        gamblerUsername: "placeholder",
        gamblerEmail: "placeholder@example.com"
    )
}

File: Android/pool/src/main/java/com/felipearpa/tyche/pool/managegamblers/PoolMemberModel.kt
package com.felipearpa.tyche.pool.managegamblers

import com.felipearpa.tyche.data.pool.domain.PoolMember
import java.util.UUID

data class PoolMemberModel(
    val gamblerId: String,
    val gamblerUsername: String,
    val gamblerEmail: String,
    val isOwner: Boolean = false,
)

fun PoolMember.toPoolMemberModel() =
    PoolMemberModel(
        gamblerId = this.gamblerId,
        gamblerUsername = this.gamblerUsername,
        gamblerEmail = this.gamblerEmail,
        isOwner = this.isOwner,
    )

fun poolMemberPlaceholderModel() =
    PoolMemberModel(
        gamblerId = UUID.randomUUID().toString(),
        gamblerUsername = "placeholder",
        gamblerEmail = "placeholder@example.com",
    )
